### ISSUE TITLE
Namesteal works inconsistently on bots

### ISSUE THREAD

**Issue by Auditormadness9** [nullworks/cathook#1440]
## System Information
**OS:** Linux
**DE:** CathookOS
**Cathook Version:** 
![image](https://user-images.githubusercontent.com/71278545/117126384-bc97a600-adab-11eb-9b82-5b178bc78c23.png)

## Config used

[Luntik.conf.zip](https://github.com/nullworks/cathook/files/6426252/Luntik.conf.zip)

## Description  
Regardless whether their namesteal option is set to Passive or Active, their names only update in the IPC webpanel, but not in-game. The only time they successfully use a stolen name in the game is at the beginning of a match (and only when they have first been booted), otherwise it's really inconsistent and works like 1% of the time, unlike for a person (for example me, I tested the option and it works normally for me). Their names don't update mid-game, so I tried the retry command, but it still did not help, they joined with their original steam names (or if i set a custom name for them, custom names). It has something to do with how bots work since the option is working fine I think (it's probably just me thinking) since the bots also can't perform cat_join_spam while a human cathooker can. I just don't understand the fact that IPC clearly shows how they change their names, but in-game they stay in their same steam names, regardless of retry or disconnect commands.

Update: okay so apparently it sometimes does the same for me, so it's not exclusive to bots I see, ignore my statement about it being the bots and not the option.

## Steps to reproduce  
- [✓] I am able to consistently reproduce this error

1. Run cat set name.namesteal 1 or 2 in the IPC.
2. Wait for bots to steal names.
3. See that bots update their names in the IPC to a stolen name.
4. Alt tab into the match, check if they have changed their names mid-game.
5. If not, try the retry command. Check that it didn't change anything.
6. Wait for 10 mins since that's the name change cooldown of the game.
7. See that nothing changed.
8. Try cat_name_set command.
9. See that nothing changed.

## Additional context 

Maybe you should consider making a bot or a user never change their names back to their original names unless and only unless namesteal is on 0 (disabled). If it is passive or active, regardless of it being in main menu or retrying to join, it should preserve the stolen name and never revert back unless told so. But even then, that could not be the problem, just a suggestion just in case that's the problem.
## Screenshot of cheats enabled  

Only Cathook.

## Checklist  
- [✓] I have included my DE, WM, and version of Cathook.
- [✓] I have included the steps to reproduce and my logs to the best I am able to share.
- [✓] The Developers are allowed to contact me on [the official telegram](https://t.me/nullworks) for further information if they need to. (Put your username here)

**Comment by BenCat07** [comment]
fix: reconnect after name steal and keep the name saved instead of clearing it

**Comment by Auditormadness9** [comment]
> fix: reconnect after name steal and keep the name saved instead of clearing it

So I was right regarding that it was clearing the stolen name when performing retry?

**Comment by BenCat07** [comment]
yes

**Comment by Auditormadness9** [comment]
Also I have tested it on myself and most of the time it changes my name mid game without reconnecting, but bots can't seem to do that, which isn't related to name resetting when reconnect.

**Comment by BenCat07** [comment]
Bots can't change their name mid-game due to the f2p changes

**Comment by Auditormadness9** [comment]
> Bots can't change their name mid-game due to the f2p changes

I have 3 p2p bots.

**Comment by Auditormadness9** [comment]
I saw a recent update (May 5), has it been fixed?

**Comment by Auditormadness9** [comment]
Just because of name change?

**Comment by BenCat07** [comment]
to change the name we need to reconnect

**Comment by Auditormadness9** [comment]
Yes but not always, right? I see sometimes in-game changes work (sometimes).

Maybe the code can read and verify if the name change actually affected the scoreboard name, if not, reconnect, if yes, no.

**Comment by BenCat07** [comment]
No, you need to either change maps, have the name upon join, or reconnect.

**Comment by Auditormadness9** [comment]
I swear I saw some of my p2p bots keep changing names in a certain server after 10 min intervals, they kept cycling and it really happened, but it only happens very rarely on certain, very certain servers.

**Comment by BenCat07** [comment]
those can obviously. Not f2p bots though.

**Comment by BenCat07** [comment]
Other name changing issues are caused by the 10 min interval.

**Comment by Auditormadness9** [comment]
> those can obviously. Not f2p bots though.

Yes, but rarely. Most of the times the p2p ones can't either.

**Comment by Auditormadness9** [comment]
> No. The biggest issue with this whole change is the reconnect possibly getting you booted for ad-hoc connections.

Just a thought, maybe not auto reconnect? Do retry manually? Also even if auto reconnect was implemented in it, adhoc limit is like 15 disconnects I think, so 15*10 = 150 minutes of playtime, which is totally ok by me.

**Comment by BenCat07** [comment]
for p2p bots a reconnect will not fix the name

**Comment by BenCat07** [comment]
also you cannot disconnect more than twice or thrixe, do not recall.

**Comment by Auditormadness9** [comment]
> for p2p bots a reconnect will not fix the name

I am currently playing and I see some p2p bots keep changing names ever 5-10 mins really fast, while mine don't, some even rollback to their steam names

**Comment by Auditormadness9** [comment]
Also noticed that sometimes a specific bot can namesteal normally every 10 mins, while others can't.

**Comment by Auditormadness9** [comment]
Any update?

**Comment by WheeledCord** [comment]
*
OS: CathookOS
DE: KDE Plasma
